docs/configuration.md

```diff
@@ -1106,7 +1106,7 @@ Apart from these, the following properties are also available, and may be useful
     parameter allows an application to run multiple executors on the
     same worker, provided that there are enough cores on that
     worker. Otherwise, only one executor per application will run on
-    each worker.
+    each worker in most cases.
```

Comment:
I think we should describe when we will break this assumption, either we treat it a bug and try to fix that, or we explicitly document the behavior. I don't think it makes much sense to just add some vague restriction such as "in most cases".